src/pocketmine/block/Chest.php

```diff
@@ -77,8 +78,26 @@ public function onPostPlace() : void{
 				if($c instanceof Chest and $c->isSameType($this) and $c->facing === $this->facing){
 					$pair = $this->world->getTile($c);
 					if($pair instanceof TileChest and !$pair->isPaired()){
-						$pair->pairWith($tile);
-						$tile->pairWith($pair);
+						if($this->facing === Facing::NORTH or $this->facing === Facing::EAST){
+							if((($c->x + ($c->z << 15)) > ($this->x + ($this->z << 15)))){
+								$ev = new ChestPairEvent($c, $this);
+							}else{
+								$ev = new ChestPairEvent($this, $c);
+							}
+						}else{
+							if((($c->x + ($c->z << 15)) < ($this->x + ($this->z << 15)))){
+								$ev = new ChestPairEvent($c, $this);
+							}else{
+								$ev = new ChestPairEvent($this, $c);
+							}
+						}
+
+						$ev->call();
+						if(!$ev->isCancelled()){
+							$pair->pairWith($tile);
+							$tile->pairWith($pair);
+						}
+
```

Comment:
I think there is a simpler way to implement this logic. A chest to the clockwise of the chest's `facing` is on its left, and anticlockwise to its right.